### PR TITLE
Fix the molding functions

### PR DESCRIPTION
**Pull request by Lyxal**
Goes towards closing #1119

### CODE REVIEW

**Review by cgccuser — APPROVED** [review]
I'm too smol brain to understand this code but the tests for this command seem to pass so LGTM

**Comment by Lyxal** [comment]
> you could copy this from jelly. it's just 9 lines [https://github.com/DennisMitchell/jellylanguage/blob/master/jelly/interpreter.py#L578-L586](https://github.com/DennisMitchell/jellylanguage/blob/master/jelly/interpreter.py?rgh-link-date=2022-06-24T14%3A31%3A14Z#L578-L586)

Mutability and lazylists say otherwise.

**Comment by cgccuser** [comment]
Wasn't it already copied from Jelly and then modified?

**Comment by Lyxal** [comment]
> Wasn't it already copied from Jelly and then modified?

That's right. In theory, it should still be the same(ish) algorithm just better suited for LazyLists and immutability.

File: vyxal/helpers.py [hunk 1]
```diff
@@ -696,15 +702,25 @@ def mold_without_repeat(
...
-        if isinstance(item, (int, str)):
+        if isinstance(item, str) or is_sympy(item):
             item = [item]
         for _ in item:
-            obj = next(content, None)
-            if obj is None:
-                break
-            temp.append(obj)
+            if type(_) in [list, LazyList]:
+                temp = mold(content[: len(_)], _)
```
Comment (chunkybanana) [inline vyxal/helpers.py:709]: Should this not be recursing mold_without_repeat?

File: vyxal/helpers.py [hunk 2]
```diff
@@ -653,23 +653,29 @@ def mold(
     The content, molded into the shape.
     """
     final = []
-    original, content = itertools.tee(content)
+    original, content = map(LazyList, itertools.tee(content))
```
Comment (chunkybanana) [inline vyxal/helpers.py:656]: This will only map at depth 1.

**Comment by Lyxal** [comment]
Replaced by #1167

**Pull request closed without merging**